pkg/arvo/sys/vane/iris.hoon

```diff
@@ -349,39 +386,73 @@
         outbound-duct.state.ax       duct
     ==
     ::
-    [moves light-gate]
+    [moves iris-gate]
   ::
       %request
     =^  moves  state.ax  (request:client +.task)
-    [moves light-gate]
+    [moves iris-gate]
   ::
       %cancel-request
     =^  moves  state.ax  cancel:client
-    [moves light-gate]
+    [moves iris-gate]
   ::
       %receive
     =^  moves  state.ax  (receive:client +.task)
-    [moves light-gate]
+    [moves iris-gate]
   ==
 ::  http-client issues no requests to other vanes
 ::
 ++  take
   |=  [=wire =duct dud=(unit goof) sign=*]
-  ^-  [(list move) _light-gate]
+  ^-  [(list move) _iris-gate]
   ~>  %spin.[%take]
   ?<  ?=(^ dud)
   !!
 ::
-++  light-gate  ..$
+++  iris-gate  ..$
 ::  +load: migrate old state to new state (called on vane reload)
 ::
 ++  load
-  |=  old=axle
-  ^+  ..^$
+  =>  |%
+      +$  axle-any
+        $%  [date=%~2019.2.8 state=state-0]
+            [date=%~2025.7.17 =state]
+        ==
+      ::
+      +$  state-0
+        $:  next-id=@ud
+            connection-by-id=(map @ud [=duct in-progress-http-request=in-progress-http-request-0])
+            connection-by-duct=(map duct @ud)
+            outbound-duct=duct
+        ==
+      +$  in-progress-http-request-0
+        $:  remaining-redirects=@ud
+            remaining-retries=@ud
+            response-header=(unit response-header:http)
+            chunks=(list octs)
+            bytes-read=@ud
+            expected-size=(unit @ud)
+        ==
+      --
+  |=  old=axle-any
+  ^+  iris-gate
   ~>  %spin.[%load]
-  ::
-  ~!  %loading
-  ..^$(ax old)
+  ?-    -.old
+      %~2019.2.8
+    %=  $
+      date.old  %~2025.7.17
+    ::
+      connection-by-id.state.old
+    %-  ~(run by connection-by-id.state.old)
+    |=  [d=duct r=in-progress-http-request-0]
+    ^-  [duct in-progress-http-request]
+    :-  d
+    :: set remaining redirects to 0 because we don't have the original request
+    [0 +:r(|5 [|5 [%'GET' '' ~ ~]]:r)]
```

Comment:
```suggestion
    ::  set remaining redirects to 0 because we don't have the original request.
    ::  it's safe to bunt the .request because it only gets used if
    ::  .remaining-redirects is non-zero.
    ::
    :-  remaining-redirects=0
    +.r(expected-size [expected-size.r *request:http])
```